### ISSUE TITLE
Nandense layer for missing values

### ISSUE THREAD

**Issue by mik3githubber** [pytorch/pytorch#87076]
### 🚀 The feature, motivation and pitch

This paper has an interesting Tensorflow implementation of a nan aware layer so that the neural network can handle missing values natively. Code provided in the paper, might be interesting to implement in Pytorch for inclusion in this library.

https://arxiv.org/pdf/2206.01640.pdf

**Comment by cpuhrsch** [comment]
This seems like a potentially good application for [MaskedTensor](https://pytorch.org/docs/master/masked.html). cc @george-qi
